### ISSUE TITLE
refactor D2L pytorch notebook code

### ISSUE THREAD

**Issue by murphyk** [probml/pyprobml#355]
Many of the colab notebooks in the supplementary material for the deep learning chapters (13-15) use pytorch code taken from [dive into deep learning](https://d2l.ai). This code is imported and used as follows:
`!wget https://raw.githubusercontent.com/d2l-ai/d2l-en/master/d2l/torch.py -q -O d2l.py`
`import d2l`
`net = d2l.foo(...)`
All this code needs refactoring.

- Copy and paste the relevant functions from [torch.py](https://github.com/d2l-ai/d2l-en/blob/master/d2l/torch.py) into the colab notebook, so the colab is self contained, and runs without importing d2l. (Note that in many cases,  the prefix d2l should be changed to torch or omitted. eg in [accuracy()](https://github.com/d2l-ai/d2l-en/blob/master/d2l/torch.py#L221), the code uses `d2l.argmax`, but it should be changed to   `torch.argmax`. Similarly, [plot_images()](https://github.com/d2l-ai/d2l-en/blob/master/d2l/torch.py#L174) has `d2l.plt.subplots`, this should be `plt.subplots`. This is because of [this line](https://github.com/d2l-ai/d2l-en/blob/master/d2l/torch.py#L24) in torch.py, which defines the symbol `d2l` to match the name of the file, namely `torch`.) 

- Some of these functions have names that only make sense in the context of the D2L book (eg `train_ch3` for the training function defined in chapter 3).  Rename the functions to be just `train`, removing hard-coded chapter numbers.

- Simplify the training code where appropriate, so it is customized for the model being trained in each colab, rather than having a bunch of if statements to handle cases that occur in later chapters. eg [train_batch_ch13](https://github.com/d2l-ai/d2l-en/blob/master/d2l/torch.py#L1520) has a line that refers to BERT; this code is not needed in the CNN colabs and might confuse the reader.
 
- Remove any #@save tags

- be sure to call torch.manual_seed(1) after np.random.seed(1) at the top of the notebook, to ensure reproducibility

**Comment by animesh-007** [comment]
Working on this.

**Comment by murphyk** [comment]
Please first make a new  issue for each notebook that needs refactoring, then tackle them one by one, so each PR has a single file in it.

**Comment by animesh-007** [comment]
> Please first make a new issue for each notebook that needs refactoring, then tackle them one by one, so each PR has a single file in it.

@murphyk I am opening new issues for each notebook. I am also seeing some notebooks, not from the chapter (13-15) using d2l pytorch code. Should I also raise issues for those notebooks too?

**Comment by murphyk** [comment]
@animesh-007  Yes, please raise issues for all d2l notebooks. (I forgot to link some of them in the markdown, but I will fix this.)

**Comment by Nirzu97** [comment]
@animesh-007  Hey, Can I also work on some of these files? which you have not worked on yet. you can give me a list of filenames or issue numbers you have opened on which I can work.

**Comment by animesh-007** [comment]
> @animesh-007 Hey, Can I also work on some of these files? which you have not worked on yet. you can give me a list of filenames or issue numbers you have opened on which I can work.

@Nirzu97 Sure. I have opened issues for all the notebooks. You can work on any of them, just put a comment on the issue that you are working on.

**Comment by animesh-007** [comment]
@Nirzu97 Thank you for contributing with me on this issue. 
@murphyk Let us know if there is any other notebook left for refactoring or not.

**Comment by murphyk** [comment]
I think this is all  finished - thanks!
